pkg/apis/kops/validation/validation.go

```diff
@@ -907,6 +924,47 @@ func validateNetworking(cluster *kops.Cluster, v *kops.NetworkingSpec, fldPath *
 		allErrs = append(allErrs, validateCIDR(cidr, fldPath.Child("additionalNetworkCIDRs").Index(i), &networkCIDRs)...)
 	}
 
+	var nonMasqueradeCIDRs []*net.IPNet
+	{
+		if v.NonMasqueradeCIDR == "" {
+			if providerConstraints.requiresNonMasqueradeCIDR {
+				allErrs = append(allErrs, field.Required(fldPath.Child("nonMasqueradeCIDR"), "Cluster does not have nonMasqueradeCIDR set"))
+			}
+		} else {
+			allErrs = append(allErrs, validateCIDR(v.NonMasqueradeCIDR, fldPath.Child("nonMasqueradeCIDR"), &nonMasqueradeCIDRs)...)
+
+			if strings.Contains(v.NonMasqueradeCIDR, ":") && v.NonMasqueradeCIDR != "::/0" {
```

Comment:
Nit: this always sets off my spidey-sense, because I think we've all been burned by this comparison many times!  we should create an IsIPv6 helper (and ideally we would use net.CIDR instead of string comparison).  Not a blocker though